org.eclipse.winery.common/src/main/java/org/eclipse/winery/common/edmm/EdmmConverter.java

```diff
@@ -0,0 +1,281 @@
+            interfaces.getInterface().forEach(anInterface -> {
+                anInterface.getOperation().forEach(operation -> {
+                    EntityId operationsEntityId = nodeTypeEntityId.extend(DefaultKeys.OPERATIONS);
+                    entityGraph.addEntity(new MappingEntity(operationsEntityId, entityGraph));
+
+                    TNodeTypeImplementation implementation = nodeTypeImplementations.values().stream()
+                        .filter(impl -> impl.getNodeType().equals(type.getQName()))
+                        .findFirst()
+                        .orElse(null);
+                    String path = getImplementationForOperation(implementation, anInterface.getName(), operation.getName());
+
+                    EntityId operationId = operationsEntityId.extend(operation.getName());
+                    entityGraph.addEntity(new ScalarEntity(
+                        path != null && this.useAbsolutePaths ? Environments.getRepositoryRoot() + "/" + path : path,
+                        operationId,
+                        entityGraph
+                    ));
+                });
+            });
+        }
+    }
+
+    private String getImplementationForOperation(TEntityTypeImplementation implementation,
+                                                 String interfaceName, String operationName) {
+        if (implementation != null && implementation.getImplementationArtifacts() != null) {
+            List<TImplementationArtifacts.ImplementationArtifact> artifacts = implementation.getImplementationArtifacts()
+                .getImplementationArtifact().stream()
+                .filter(artifact -> artifact.getInterfaceName().equals(interfaceName))
+                .collect(Collectors.toList());
+
+            if (artifacts.size() == 1 && artifacts.get(0).getArtifactRef() != null) {
+                TArtifactTemplate artifactTemplate = artifactTemplates.get(artifacts.get(0).getArtifactRef());
+                if (artifactTemplate.getArtifactReferences().getArtifactReference().size() > 0) {
+                    return artifactTemplate.getArtifactReferences().getArtifactReference().get(0).getReference();
+                }
+            }
+
+            for (TImplementationArtifacts.ImplementationArtifact artifact : artifacts) {
+                if (artifact.getOperationName().equals(operationName)) {
+                    TArtifactTemplate artifactTemplate = artifactTemplates.get(artifact.getArtifactRef());
+                    if (artifactTemplate != null && artifactTemplate.getArtifactReferences().getArtifactReference().size() > 0) {
+                        return artifactTemplate.getArtifactReferences().getArtifactReference().get(0).getReference();
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    private String normalizeQName(QName qName) {
+        return qName.toString().substring(1)
+            .replace("}", "__")
```

Comment:
What about `{`?

Reply:
--> `substring(1)`

Reply:
Oh wow, and you are sure that the first char is ALWAYS a `{`? :-)